File: Algorithms/Sorting/BubbleSort/Java/BubbleSort.java
class BubbleSort {

	void bubbleSort(int arr[])
	{
		int len = arr.length;
		for (int foo = 0; foo < len; foo++)
			for (int i = 0; i<len-1; i++)
				if (arr[i+1]<arr[i])
				{
					int tmp = arr[i];
					arr[i] = arr[i+1];
					arr[i+1] = tmp;
				}
	}

	void printArray(int arr[])
	{
		int len = arr.length;
		for (int i=0; i<len; ++i)
		{
			System.out.print(arr[i]);
			if(i<len-1)
				System.out.print(", ");
		}
		System.out.println();
	}

	// main func
	public static void main(String args[])
	{
		System.out.println("Bubble Sort!");
		
		BubbleSort bs = new BubbleSort();
		int arr[] = {1,5,2,0,88,12,3};

		System.out.println("Original array");
		bs.printArray(arr);
		
		bs.bubbleSort(arr);
		System.out.println("Sorted array");
		bs.printArray(arr);
	}
}

File: Algorithms/Sorting/BubbleSort/CPP/bubble_sort.cpp
//============================================================================
// Name        : bubble_sort.cpp
// Author      : 
// Version     :
// Copyright   : Your copyright notice
// Description : Hello World in C++, Ansi-style
//============================================================================

#include <iostream>
#include <vector>
using namespace std;

vector<int> bsort(vector<int> v)
{
	int size=v.size();
	for(int foo=0;foo<size;foo++)
	{
		for(int i=0;i<size-1;i++)
		{
			if(v[i+1]<v[i])
			{
				int tmp = v[i];
				v[i]=v[i+1];
				v[i+1]=tmp;
			}
		}

	}
	return v;
}

void print_vec(vector<int> v)
{
	int size=v.size();
	for(int i=0;i<size;i++)
	{
		cout << v[i];
		if(i<size-1)
			cout << ", ";
	}
	cout << endl;

}

int main() {
	cout << "Bubble Sort" << endl; // prints !!!Hello World!!!

	vector<int> vec;
	vec.push_back(1);
	vec.push_back(5);
	vec.push_back(2);
	vec.push_back(0);
	vec.push_back(88);
	vec.push_back(12);
	vec.push_back(3);

	print_vec(vec);
	vector<int> sv=bsort(vec);
	print_vec(sv);

	return 0;
}
